src/rules/no-descending-specificity/index.js

```diff
@@ -35,6 +36,9 @@ export default function (actual) {
         // Ignore `.selector, { }`
         if (trimSelector === "") { return }
 
+        // Ignore custom property set `--foo: {};`
+        if (isCustomPropertySet(rule)) { return }
```

Comment:
Can you please move this line to between line 31 and 32. We'd want to return _as early as possible_, i.e. before doing anything to the selector.